### PR TITLE
Add stimulus_template to IntracellularStimuliTable

### PR DESCRIPTION
**Pull request by iawbrooks**
## Summary of changes

- Added the optional `stimulus_template` column to `IntracellularStimuliTable`. The purpose of this column is to store stimulus template waveforms, i.e. waveforms that were not _recorded_, but were defined as protocols in the recording software.
From the perspective of writing analysis tools, it is much easier to analyze recordings if this noiseless version of the stimulus is available.

## Checklist

For all schema changes:
- [x] Add release notes for the PR to `docs/format/source/format_release_notes.rst`.

If this is the first schema change after a schema release (i.e., the version string in `core/nwb.namespace.yaml` does not
end in "-alpha"), then:
- [ ] Update the version string in `core/nwb.namespace.yaml` and `core/nwb.file.yaml` to the next major/minor/patch
  version with the suffix "-alpha". For example, if the current version is 2.4.0 and this is a minor change, then the
  new version string should be "2.5.0-alpha".
- [ ] Update the value of the `version` variable in `docs/format/source/conf.py` to the next version **without** the
  suffix "-alpha", e.g., "2.5.0".
- [ ] Update the value of the `release` variable in `docs/format/source/conf.py` to the next version **with** the suffix
  "-alpha", e.g., "2.5.0-alpha".
- [ ] Add a new section in the release notes `docs/format/source/format_release_notes.rst` for the new version
  with the date "Upcoming" in parentheses.

### CODE REVIEW

File: docs/format/source/format_release_notes.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ Minor changes
 ^^^^^^^^^^^^^
 - Fixed typos in docstrings. (#560)
 - Modified `OptogeneticSeries` to allow 2D data, primarily in extensions of `OptogeneticSeries`. (#564)
+- Added optional ``stimulus_template`` column to ``IntracellularRecordingsTable`` (#545)
```
Comment (oruebel) [inline docs/format/source/format_release_notes.rst]:
```suggestion
- Added optional ``stimulus_template`` column to ``IntracellularStimuliTable`` as part of the ``IntracellularRecordingsTable`` (#545)
```

**Review by oruebel — DISMISSED** [review]

**Comment by oruebel** [comment]
@stephprince does this PR require updating the version of the namespace or is that already done on the main branch?

**Comment by stephprince** [comment]
the namespace version is already updated to 2.7.0-alpha on the main branch so should be ok

**Comment by oruebel** [comment]
@rly this is also ready to merge

**Review by rly — APPROVED** [review]

**Pull request merged**